module/lezioni.py

```diff
@@ -1,60 +1,202 @@
-        'terzo':3
+        '1' : 'Lunedì',
+        '2' : 'Martedì',
+        '3' : 'Mercoledì',
+        '4' : 'Giovedì',
+        '5' : 'Venerdì'
     }
 
-    return switcher.get(anno, "Anno non valido, utilizzare primo/secondo/terzo")
-
-def lezioni_output(item):
-    output = "*Nome:* " + item["nome"]
-    output += "\n*Aula:* " + str(item["aula"])
-    output += "\n*Anno:* " + str(item["anno"]) + "\n"
+    return switcher.get(day, "Giorno non valido")
 
+# eliminate
+def lezioni_button():
+    output = "Scrivi /lezioni inserendo almeno uno dei seguenti parametri: giorno, materia"
     return output
 
-def lezioni_condition(items, condition, *arg):
-    output = set()
-    for item in items:
-        if( arg and ( (arg[0] in str(item[condition]).lower()) or (get_numero_anno(arg[0]) == item[condition]) ) ): # controlla se viene passato un arg, se come arg verrà passato il nome allora il primo check sarà vero, altrimenti controlleremo se è stato passato l'anno
-            output.add(lezioni_output(item))
+def get_lezioni_text_InLineKeyboard(context: CallbackContext) -> (str, InlineKeyboardMarkup): #restituisce una tuple formata da (message_text, InlineKeyboardMarkup)
+    keyboard = [[]]
+
+    lezioni_user_data = context.user_data['lezioni']
+    text_anno = ", ".join([key for key in lezioni_user_data if "anno" in key]).replace("anno","") #stringa contenente gli anni per cui la flag è true
+    text_giorno = ", ".join([key for key in lezioni_user_data if "giorno" in key]).replace("giorno","") #stringa contenente le lezioni per cui la flag è true
+    text_insegnamento = lezioni_user_data.get("insegnamento", "") #stringa contenente l'insegnamento
+
+    message_text = "Anno: {}\nGiorno: {}\nInsegnamento: {}"\
+        .format(text_anno if text_anno else "tutti",\
+                text_giorno if text_giorno else "tutti",\
+                text_insegnamento if text_insegnamento else "tutti")
+    keyboard.append([InlineKeyboardButton(" ~ Personalizza la ricerca ~ ", callback_data="_div")])
+    keyboard.append(
+            [
+                InlineKeyboardButton(" Anno ", callback_data="sm_lezioni_button_anno"),
+                InlineKeyboardButton(" Giorno ", callback_data="sm_lezioni_button_giorno"),
+            ]
+        )
+    keyboard.append(
+            [
+                InlineKeyboardButton(" Insegnamento ", callback_data="sm_lezioni_button_insegnamento"),
+                InlineKeyboardButton(" Cerca ", callback_data="lezioni_button_search")
+            ]
+        )
+
+    return message_text, InlineKeyboardMarkup(keyboard)
+
+def lezioni(update: Update, context: CallbackContext):
+    check_log(update, context, "lezioni")
+    if 'lezioni' in context.user_data: context.user_data['lezioni'].clear() #ripulisce il dict dell'user relativo al comando /esami da eventuali dati presenti
+    else: context.user_data['lezioni'] = {} #crea il dict che conterrà i dati del comando /esami all'interno della key ['esami'] di user data
+
```

Comment:
Some comments still refer to the /esami command. You may want to change that, for better clarity